pkg/storage/storage.go

```diff
@@ -111,42 +107,22 @@ func NewDriver(crname string, crnamespace string, cfg *opapi.ImageRegistryConfig
 // getPlatformStorage returns the storage type that should be used
 // based on the cloudplatform we are running on, as determined
 // from the installer configuration.
-func getPlatformStorage() (string, error) {
-	cm := &corev1.ConfigMap{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: corev1.SchemeGroupVersion.String(),
-			Kind:       "ConfigMap",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "cluster-config-v1",
-			Namespace: "kube-system",
-		},
-	}
-
-	if err := sdk.Get(cm); err != nil {
-		return "", fmt.Errorf("unable to read cluster install configuration: %v", err)
-	}
-
-	installConfig := &installer.InstallConfig{}
-	if err := yaml.NewYAMLOrJSONDecoder(strings.NewReader(cm.Data["install-config"]), 100).Decode(installConfig); err != nil {
-		return "", fmt.Errorf("unable to decode cluster install configuration: %v", err)
+func getPlatformStorage() (opapi.ImageRegistryStorageType, error) {
+	installConfig, err := oputil.GetInstallConfig()
+	if err != nil {
+		return "", err
 	}
 
 	// if we can't determine what platform we're on, fallback to creating
 	// a PVC for the registry.
-	storage := ""
 	switch {
 	case installConfig.Platform.Libvirt != nil:
-		storage = "emptydir"
-		// TODO - Corey enable s3, someone enable swift in the future.
-		/*
-			case installConfig.Platform.AWS != nil:
-				storage = "s3"
-			case installConfig.Platform.OpenStack != nil:
-				storage = "swift"
-			default:
-			  	storage="filesystem"
-		*/
+		return opapi.StorageTypeEmptyDir, nil
+	case installConfig.Platform.AWS != nil:
+		return opapi.StorageTypeS3, nil
+		// case installConfig.Platform.OpenStack != nil:
+		// 	return opapi.StorageTypeSwift, nil
+	default:
+		return opapi.StorageTypeFileSystem, nil
```

Comment:
we can't start returning type FileSystem until the rest of the wiring actually does the right thing with it. comment this out.

and put the commented out block for Swift back in.

Reply:
(make `default:` return emptystring for now)

Reply:
Updated in next push